### ISSUE TITLE
Can't pull Containers

### ISSUE THREAD

**Issue by SonicFM** [fkie-cad/FACT_docker#22]
Branch: fact4.0

First of all thanks for providing FACT_core as well as FACT_docker!

I'm trying to setup FACT_docker with Branch fact4.0 as described in "Usage", but sadly i already failed at the first command.

docker pull ghcr.io/fkie-cad/fact-core-frontend:4.0
Error response from daemon: Head "https://ghcr.io/v2/fkie-cad/fact-core-frontend/manifests/4.0": denied
and
./start.py pull
Unable to find image 'fkie-cad/fact-core-scripts:4.0' locally
docker: Error response from daemon: pull access denied for fkie-cad/fact-core-scripts, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.

What am i missing here? Or is it even my fault?

Is there an projected timeline when FACT will be available over docker hub?
As stated in https://github.com/fkie-cad/FACT_docker/pull/20 there is https://hub.docker.com/search?q=maringuu. Should i try these instead?

Thanks in advance!

**Comment by maringuu** [comment]
Hello! Thank you for your interest in FACT and FACT_docker.
The problem is that the docker images are still not released by the fkie-cad account. Once the PR is merged the images will be released.
> As stated in https://github.com/fkie-cad/FACT_docker/pull/20 there is https://hub.docker.com/search?q=maringuu. Should i try these instead?

Exactly right! These are the images that are build by github actions on my fork. They are equivalent to the images pointed out in the readme, since the PR is ready and only waits for a final review.
Note that the scripts also have to be adpted to use the _right_ images:
```sh
sed -i s/fkie-cad/maringuu/ docker-compose.yml
sed -i s/fkie-cad/maringuu/ start.py
```

Closing this issue as the problem is hopefully solved and is about a development branch.
Feel free to reopen if it still does not work.

If you encounter anything that does not work as expected we are happy if you tell us! So if you have the time to spare just comment on the PR. Thanks!

**Comment by SonicFM** [comment]
Hello maringuu,
thank you for your quick reply!

Changing the docker-compose.yml and the start.py worked, thanks a lot!

But I have another error, even if I know where it comes from I unfortunately do not know how to fix it.

I use docker namespace remapping the problems inside the containers which are in the docker-compose.yml I could solve by setting " userns_mode: "host" " for each container.

However, start.py loads the fact extraction container, but this fails because it seems to start without the "userns_mode: "host" " parameter. See also https://github.com/fkie-cad/FACT_core/issues/794. This looks very much like the same problem.

```
./start.py pull

[2022-07-29 13:33:41][install][INFO]: FACT Installer 1.2
[2022-07-29 13:33:41][install][INFO]: Your distribution (alpine 3.15.5) is not supported. FACT Installer requires Ubuntu 18.04, 20.04 or compatible!
[2022-07-29 13:33:41][backend][INFO]: Pulling fact extraction container
Traceback (most recent call last):
  File "/opt/FACT_core/src/install.py", line 187, in <module>.
    install()
  File "/opt/FACT_core/src/install.py", line 156, in install
    install_docker_images(args)
  File "/opt/FACT_core/src/install.py", line 180, in install_docker_images
    backend_install_docker_images()
  File "/opt/FACT_core/src/install/backend.py", line 80, in _install_docker_images
    raise InstallationError(f'Failed to pull extraction container:\n{docker_process.stdout}')
helperFunctions.install.InstallationError: Failed to pull extraction container:
Using default tag: latest
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/create?fromImage=fkiecad%2Ffact_extractor&tag=latest": dial unix /var/run/docker.sock: connect: permission denied
```

Is there any way I can solve this myself/locally? I'm sure it wouldn't hurt to find a "public" solution for this as well, since others would have the same problem using docker's namespace remapping.

Thanks again!

**Comment by maringuu** [comment]
Hmm I haven't investigated much but is it possible that `--group-add` does not work in combination with usernamespaces? The `start.py` scripts adds the gid of the docker host this way. Maybe this does not work.
Please open another issue to track this. This problem is seperate from this one.

**Comment by cosad3s** [comment]
> Hello! Thank you for your interest in FACT and FACT_docker. The problem is that the docker images are still not released by the fkie-cad account. Once the PR is merged the images will be released.
> 
> > As stated in #20 there is https://hub.docker.com/search?q=maringuu. Should i try these instead?
> 
> Exactly right! These are the images that are build by github actions on my fork. They are equivalent to the images pointed out in the readme, since the PR is ready and only waits for a final review. Note that the scripts also have to be adpted to use the _right_ images:
> 
> ```shell
> sed -i s/fkie-cad/maringuu/ docker-compose.yml
> sed -i s/fkie-cad/maringuu/ start.py
> ```
> 
> Closing this issue as the problem is hopefully solved and is about a development branch. Feel free to reopen if it still does not work.
> 
> If you encounter anything that does not work as expected we are happy if you tell us! So if you have the time to spare just comment on the PR. Thanks!
> 
> Edit: As for the timeline I think that they might be released next week or so.

Still have the issue for information :) Thanks for the workaround

**Comment by maringuu** [comment]
@jstucke Gentle Ping

**Comment by cosad3s** [comment]
After the workaround applied, it is still not work ... After some minutes of preparation I get this error:

```
./start.py pull
[2022-10-13 07:06:56][install][INFO]: FACT Installer 1.2
[2022-10-13 07:06:56][install][INFO]: Your Distribution (alpine 3.15.6) is not supported. FACT Installer requires Ubuntu 18.04, 20.04 or compatible!
[2022-10-13 07:06:56][backend][INFO]: Pulling fact extraction container
[2022-10-13 07:09:42][install][CRITICAL]: Your Distribution (alpine 3.15.6) is not supported. FACT Installer requires Ubuntu 18.04, 20.04 or compatible!
```

I am running it on a Kali, but I think this error comes from a container right ?
```
# uname -a    
Linux cactus-ths 5.18.0-kali5-amd64 #1 SMP PREEMPT_DYNAMIC Debian 5.18.5-1kali6 (2022-07-07) x86_64 GNU/Linux
```

**Comment by maringuu** [comment]
>  After some minutes of preparation I get this error

See #24, you can ignore the error message.

Also the images have been published.

**Comment by cosad3s** [comment]
Thanks. Authentication is needed ?

I followed https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry and add `read:packages` scope for a new token.

```bash
# docker pull ghcr.io/fkie-cad/fact-core-frontend:4.0.1          
Error response from daemon: denied
```

**Comment by maringuu** [comment]
> Thanks. Authentication is needed ?

Whoops, we forgot to make the images public... The are now available. So just follow the instructions in the README and you are good to go

**Comment by infinite2005qian** [comment]
> Collaborator

how you solved it?

**Comment by infinite2005qian** [comment]
> Collaborator

how you solved it?

> After the workaround applied, it is still not work ... After some minutes of preparation I get this error:
> 
> ```
> ./start.py pull
> [2022-10-13 07:06:56][install][INFO]: FACT Installer 1.2
> [2022-10-13 07:06:56][install][INFO]: Your Distribution (alpine 3.15.6) is not supported. FACT Installer requires Ubuntu 18.04, 20.04 or compatible!
> [2022-10-13 07:06:56][backend][INFO]: Pulling fact extraction container
> [2022-10-13 07:09:42][install][CRITICAL]: Your Distribution (alpine 3.15.6) is not supported. FACT Installer requires Ubuntu 18.04, 20.04 or compatible!
> ```
> 
> I am running it on a Kali, but I think this error comes from a container right ?
> 
> ```
> # uname -a    
> Linux cactus-ths 5.18.0-kali5-amd64 #1 SMP PREEMPT_DYNAMIC Debian 5.18.5-1kali6 (2022-07-07) x86_64 GNU/Linux
> ```
> 
> Kind of conflict ? Between:
> 
> * https://github.com/fkie-cad/FACT_core/blob/master/src/helperFunctions/install.py
> * https://github.com/fkie-cad/FACT_docker/blob/main/Dockerfile-scripts

how you solved it?

**Comment by maringuu** [comment]
We should probably print a success message but as I stated above you can ignore this error.
> See https://github.com/fkie-cad/FACT_docker/issues/24, you can ignore the error message.
